### PR TITLE
MFTF 2.3.12 documentation updates

### PR DESCRIPTION
**Pull request by dshevtsov**
## This PR is a:

- Content update

## Summary

When this pull request is merged, it will update Metadata topic with more details  about referencing to data from `url` of an `<operation>` and in `<value>` of an `<array>`.

whatsnew
MFTF 2.3.12 was released:
- Added more documentation with examples for [`<value>`](https://devdocs.magento.com/mftf/2.3/metadata.html#value-tag) in metadata array.
- Added use of reference to data field in `url` of [metadata operation](https://devdocs.magento.com/mftf/2.3/metadata.html#operation-tag).

### CODE REVIEW

File: mftf/2.3/metadata.md [hunk 1]
```diff
@@ -70,11 +73,13 @@ The following diagram demonstrates the XML structure of a metadata file:
...
 ## Principles   {#principles}
 
-* The `dataType` value must match the `type` value of the corresponding entity.
-* The file name contains data type split with `_` and ends with `-meta`.
-Example: `product_attribute-meta.xml`.
-* A metadata file may contain different types of operations (`type`) with the same data entity (`dataType`).
-Example: 
+1. The `dataType` value must match the `type` value of the corresponding entity.
+2. The file name contains data type split with `_` and ends with `-meta`.
+   Example: `product_attribute-meta.xml`.
```
Comment (jilu1) [inline mftf/2.3/metadata.md]: file name can be anything but needs to ends with `-meta`. We use data type a lot by convention.

File: mftf/2.3/metadata.md [hunk 2]
```diff
@@ -405,34 +422,35 @@ Root element that points to the corresponding XML Schema.
...
+| `returnRegex`   | string                                                                       | optional | Determines if the response contains the matching value to return.                                                                            |
+| `removeBackend` | boolean                                                                      | optional | Removes backend name from requested URL. Applicable when `auth="adminFormKey"`.                                                              |
+| `filename`      | string                                                                       | optional |                                                                                                                                              |
+
+- \*`url` - full URL is a concatenation of _ENV.baseUrl_ + `/rest/` + _url_.
+  To reuse data of a required entity or returned response use a field key wrapped in curly braces such as `{sku}`.
+  When the data to reuse is of a different type, declare also the type of data such as `{product.sku}`.
+  Examples: `"/V1/products/{sku}/media/{id}"`.
```
Comment (jilu1) [inline mftf/2.3/metadata.md]: missing type? should be
Examples: `"/V1/products/{product.sku}/media/{id}"`.

File: mftf/2.3/metadata.md [hunk 3]
```diff
@@ -405,34 +422,35 @@ Root element that points to the corresponding XML Schema.
...
-`successRegex`|string|optional|Determines if the operation was successful. Parses the HTML body in response and asserts if the value assigned to the `successRegex` exists.
-`returnRegex`|string|optional| Determines if the response contains the matching value to return.
-`removeBackend`|boolean|optional|Removes backend name from requested URL. Applicable when `auth="adminFormKey"`.
-
-{% include note.html
-type="info"
-content="GET is used for retrieving data from objects.<br/>
-POST is used for creating new objects.<br/>
-PUT is used for updating objects.<br/>
-DELETE is used for deleting objects.<br/>" %}
```
Comment (jilu1) [inline mftf/2.3/metadata.md]: DELETE is removed. Is this on purpose?

File: mftf/2.3/metadata.md [hunk 4]
```diff
@@ -441,43 +459,94 @@ Sets one of the following operation types:
...
+| Attribute | Type   | Use      | Description        |
+| --------- | ------ | -------- | ------------------ |
+| `key`     | string | required | Name of the array. |
 
 It contains one or more `value` elements.
 
 ### value {#value-tag}
 
-An item in `array`.
+Declares a data type of the parent array items.
```
Comment (jilu1) [inline mftf/2.3/metadata.md]: It's the array itself. Not parent array.

File: mftf/2.3/metadata.md (hunk 2 above)
Comment (dshevtsov) [reply mftf/2.3/metadata.md]: :ok:

File: mftf/2.3/metadata.md (hunk 4 above)
Comment (dshevtsov) [reply mftf/2.3/metadata.md]: :ok:

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=3431) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=3431) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: dshevtsov<br/>:white_check_mark: jilu1<br/>:x: Donald Booth<hr/>**Donald Booth** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

File: mftf/2.3/metadata.md [hunk 5]
```diff
@@ -9,15 +9,16 @@ _This topic was updated due to the {{page.mftf-release}} MFTF release._
 {: style="text-align: right"}
 
 In this topic we talk about handling entities that you need in your tests (such as categories, products, wish lists, and similar) using the MFTF.
-Using data handling actions like [createData], [deleteData], [updateData], and [getData], you are able to create, delete, update, and read entities for your tests. 
+Using data handling actions like [createData], [deleteData], [updateData], and [getData], you are able to create, delete, update, and read entities for your tests.
```
Comment (lorikrell) [inline mftf/2.3/metadata.md]: Should these be in ticks for code names? Example edit:

```suggestion
Using data handling actions like [`createData`], [`deleteData`], [`updateData`], and [`getData`], you are able to create, delete, update, and read entities for your tests.
```

File: mftf/2.3/metadata.md [hunk 6]
```diff
@@ -441,43 +459,94 @@ Sets one of the following operation types:
...
+| `key`     | string | required | Name of the array. |
 
 It contains one or more `value` elements.
 
 ### value {#value-tag}
 
-An item in `array`.
+Declares a data type for items within `<array>`.
+
+1. Example of an array with value of a primitive data type:
```
Comment (lorikrell) [inline mftf/2.3/metadata.md]: General question, should these be in a numbered list? Is there a specific task they are completing with the steps? Or you can remove the numbering and keep the bullets under each example.

**Comment by dobooth** [comment]
running tests

**Pull request merged**